src/quote/Quote.ts

```diff
@@ -0,0 +1,191 @@
+      cTypeHash: deserializedQuote.cTypeHash,
+      cost: deserializedQuote.cost,
+      currency: deserializedQuote.currency,
+      quoteTimeframe: deserializedQuote.quoteTimeframe,
+      termsAndConditions: deserializedQuote.termsAndConditions,
+      version: deserializedQuote.version,
+    })
+    if (!Quote.validateQuoteSchema(QuoteSchema, quote)) {
+      throw new Error('Quote does not correspond to schema')
+    }
+    if (!Quote.verifyQuoteHash(quote, deserializedQuote.quoteHash)) {
+      throw Error('Invalid Quote Hash')
+    }
+    if (
+      !verify(
+        JSON.stringify(quote),
+        deserializedQuote.attesterSignature,
+        deserializedQuote.attesterAddress
+      )
+    ) {
+      throw Error(
+        `attestersSignature ${deserializedQuote.attesterSignature}
+        does not check out with the supplied data`
+      )
+    }
+    return {
+      attesterAddress: quote.attesterAddress,
+      cTypeHash: quote.cTypeHash,
+      cost: quote.cost,
+      currency: quote.currency,
+      quoteTimeframe: quote.quoteTimeframe,
+      termsAndConditions: quote.termsAndConditions,
+      version: quote.version,
+      quoteHash: deserializedQuote.quoteHash,
+      attesterSignature: deserializedQuote.attesterSignature,
+    }
+  }
+
+  public static fromQuoteDataAndIdentity(
+    quoteInput: IQuote,
+    identity: Identity
+  ): IQuoteAttesterSigned {
+    if (!Quote.validateQuoteSchema(QuoteSchema, quoteInput)) {
+      throw new Error('Quote does not correspond to schema')
+    }
+    const quote = new Quote(quoteInput)
+    return quote.createAttesterSignature(identity)
+  }
+
+  public attesterAddress: IQuote['attesterAddress']
+  public cTypeHash: IQuote['cTypeHash']
+  public cost: IQuote['cost']
+  public currency: IQuote['currency']
+  public quoteTimeframe: IQuote['quoteTimeframe']
+  public termsAndConditions: IQuote['termsAndConditions']
+  public version: IQuote['version']
+
+  public constructor(quoteInput: IQuote) {
+    this.attesterAddress = quoteInput.attesterAddress
+    this.cTypeHash = quoteInput.cTypeHash
+    this.cost = quoteInput.cost
+    this.currency = quoteInput.currency
+    this.quoteTimeframe = quoteInput.quoteTimeframe
+    this.termsAndConditions = quoteInput.termsAndConditions
+    this.version = quoteInput.version
+  }
+
+  public createAttesterSignature(
+    attesterIdentity: Identity
+  ): IQuoteAttesterSigned {
+    const generatedQuoteHash = hashObjectAsStr(this)
+    const Quotetemp = {
+      attesterAddress: this.attesterAddress,
+      cTypeHash: this.cTypeHash,
+      cost: this.cost,
+      currency: this.currency,
+      quoteTimeframe: this.quoteTimeframe,
+      termsAndConditions: this.termsAndConditions,
+      version: this.version,
+      quoteHash: generatedQuoteHash,
+    }
+    const signature = attesterIdentity.signStr(JSON.stringify(Quotetemp))
+    if (!Quote.verifyQuoteHash(this, generatedQuoteHash)) {
+      throw Error('Invalid Quote Hash')
+    }
+    return {
```

Comment:
please use object spread, or similar to make it more compact

Reply:
done